### PR TITLE
[ET-857] [ET-890] [ET-891] [ET-892] Repairs for Opt-in and Can't Go

### PR DESCRIPTION
**Pull request by sc0ttkclark**
[ET-857]
[ET-890]
[ET-891]
[ET-892]

Screencast: https://share.skc.dev/wbu7eqQR

* Save the correct order status for attendee by getting it correctly from attendee details
* Pass order status from success step into template
* Check order status for whether to show opt-in field, also add filter to hide the option
* Default optout to 1

[ET-857]: https://moderntribe.atlassian.net/browse/ET-857
[ET-890]: https://moderntribe.atlassian.net/browse/ET-890
[ET-891]: https://moderntribe.atlassian.net/browse/ET-891
[ET-892]: https://moderntribe.atlassian.net/browse/ET-892

### CODE REVIEW

File: src/views/v2/rsvp/actions/success/toggle.php [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+ */
+
+if ( $opt_in_toggle_hidden ) {
+	return;
+}
+
+$toggle_id = 'toggle-rsvp-' . $rsvp->ID;
+?>
+
+<div class="tribe-tickets__rsvp-actions-success-going-toggle tribe-common-form-control-toggle">
```
Comment (paulmskim) [inline src/views/v2/rsvp/actions/success/toggle.php:30]: should we add snapshot tests for this?

File: src/views/v2/rsvp/ari/form.php [hunk 2]
```diff
@@ -23,6 +23,7 @@
 
 	<input type="hidden" name="tribe_tickets[<?php echo esc_attr( absint( $rsvp->ID ) ); ?>][ticket_id]" value="<?php echo esc_attr( absint( $rsvp->ID ) ); ?>">
 	<input type="hidden" name="tribe_tickets[<?php echo esc_attr( absint( $rsvp->ID ) ); ?>][attendees][0][order_status]" value="<?php echo esc_attr( $going ); ?>">
+	<input type="hidden" name="tribe_tickets[<?php echo esc_attr( absint( $rsvp->ID ) ); ?>][attendees][0][optout]" value="1">
```
Comment (paulmskim) [inline src/views/v2/rsvp/ari/form.php:26]: is there a snapshot test that needs to be updated for this?

File: src/views/v2/rsvp/form/form.php [hunk 3]
```diff
@@ -25,6 +25,7 @@
 >
 	<input type="hidden" name="tribe_tickets[<?php echo esc_attr( absint( $rsvp->ID ) ); ?>][ticket_id]" value="<?php echo esc_attr( absint( $rsvp->ID ) ); ?>">
 	<input type="hidden" name="tribe_tickets[<?php echo esc_attr( absint( $rsvp->ID ) ); ?>][attendees][0][order_status]" value="<?php echo esc_attr( $going ); ?>">
+	<input type="hidden" name="tribe_tickets[<?php echo esc_attr( absint( $rsvp->ID ) ); ?>][attendees][0][optout]" value="1">
```
Comment (paulmskim) [inline src/views/v2/rsvp/form/form.php:28]: is there a snapshot test that needs to be updated for this?

File: src/views/v2/rsvp/actions/success/toggle.php (hunk 1 above)
Comment (sc0ttkclark) [reply src/views/v2/rsvp/actions/success/toggle.php:30]: Added

File: src/views/v2/rsvp/ari/form.php (hunk 2 above)
Comment (sc0ttkclark) [reply src/views/v2/rsvp/ari/form.php:26]: Updated

File: src/views/v2/rsvp/form/form.php (hunk 3 above)
Comment (sc0ttkclark) [reply src/views/v2/rsvp/form/form.php:28]: Updated

**Pull request merged**